Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -789,6 +794,15 @@ void Device::draw_primitives(PrimitiveType primitive_type, FloatMatrix4x4 const&
         }
     }
 
+    // Generate texture coordinates if at least one coordinate is enabled
+    bool texture_coordinate_generation_enabled = false;
+    for (size_t i = 0; i < m_options.texcoord_generation_enabled_coordinates.size(); ++i) {
```

Comment:
This can be a regular for each loop

Reply:
I need to `i` to access `m_options.texcoord_generation_enabled_coordinates[i]` maybe putting all that config per texture unit into a new struct would be better.